charts/argo-templates/templates/soft-delete-table.yaml

```diff
@@ -0,0 +1,122 @@
+            arguments:
+              parameters:
+                - name: url
+                  value: '{{ "{{inputs.parameters.url}}" }}'
+                - name: job-id
+                  value: '{{ "{{tasks.submit-soft-delete.outputs.result}}" }}'
+                - name: timeout
+                  value: '{{ "{{inputs.parameters.timeout}}" }}'
+                - name: sa-secret
+                  value: '{{ "{{inputs.parameters.sa-secret}}" }}'
+                - name: sa-secret-key
+                  value: '{{ "{{inputs.parameters.sa-secret-key}}" }}'
+    {{- $secretVolume := "sa-secret-volume" }}
+    {{- $mountPath := "/secret" }}
+    # SubTemplates
+    - name: submit-soft-delete
+      inputs:
+        parameters:
+          - name: url
+          - name: dataset-id
+          - name: source-path
+          - name: table-name
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ "{{inputs.parameters.sa-secret}}" }}'
+      script:
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: DATASET_ID
+            value: '{{ "{{inputs.parameters.dataset-id}}" }}'
+          - name: SOURCE_PATH
+            value: '{{ "{{inputs.parameters.source-path}}" }}'
+          - name: TABLE_NAME
+            value: '{{ "{{inputs.parameters.table-name}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/{{inputs.parameters.sa-secret-key}}" $mountPath }}'
+        volumeMounts:
+          - name: '{{ $secretVolume }}'
+            mountPath: '{{ $mountPath }}'
+        command: [python]
+        source: |
+          {{- range .Files.Lines "scripts/submit-table-soft-delete.py" }}
+          {{ . }}
+          {{- end }}
+    - name: poll-job
+      inputs:
+        parameters:
+          - name: url
+          - name: job-id
+          - name: timeout
+          - name: sa-secret
+          - name: sa-secret-key
+      volumes:
+        - name: '{{ $secretVolume }}'
+          secret:
+            secretName: '{{ "{{inputs.parameters.sa-secret}}" }}'
+      script:
+        image: us.gcr.io/broad-dsp-gcr-public/monster-auth-req-py:1.0.1
+        env:
+          - name: API_URL
+            value: '{{ "{{inputs.parameters.url}}" }}'
+          - name: JOB_ID
+            value: '{{ "{{inputs.parameters.job-id}}" }}'
+          - name: TIMEOUT
+            value: '{{ "{{inputs.parameters.timeout}}" }}'
+          - name: GOOGLE_APPLICATION_CREDENTIALS
+            value: '{{ printf "%s/{{inputs.parameters.sa-secret-key}}" $mountPath }}'
+        volumeMounts:
+          - name: '{{ $secretVolume }}'
+            mountPath: '{{ $mountPath }}'
+        command: [python]
+        source: |
+          {{- range .Files.Lines "scripts/poll-ingest-job.py" }}
```

Comment:
Nitpick: There's now a helper that could make this slightly less verbose, you could do:
```
source: |
{{- include "argo.render-lines" (.Files.Lines "scripts/poll-ingest-job.py") | indent 10 }}
```
Can be used for all the script sources.

Reply:
Thank you! I just updated with those changes. 

Also, a quick question on testing: is there a way to test any of these changes before this branch is merged? Or is it just a matter of merging this, and then seeing if my changes to the clinvar orchestration can be deployed without errors?

Reply:
There's no clean way to test this without merging and pulling the auto-deploy. It's super annoying, but I'm hoping it won't bite us that often in practice once we stop with these big shifts of functionality into monster-helm